ldappool/__init__.py

```
@@ -0,0 +1,506 @@
+        return True
+
+    def __authenticate__(self):
+        try:
+            self.conn.simple_bind_s(self.binddn, self.bindpw)
+            ldappool_auth_calls_success.inc()
+            logging.debug("__whoami from __authenticate__")
+            self.__whoami()
+        except ldap.INVALID_CREDENTIALS as ldaperr:
+            ldappool_auth_calls_failure.inc()
+            logging.info(ldaperr)
+            raise ldap.INVALID_CREDENTIALS
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)